Wrappers/Python/cil/framework/data_container.py

```diff
@@ -0,0 +1,1548 @@
+
+        # the sin of the linear distance gives us an approximation of area of the circle to include in the mask
+        numpy.sin(mask, out = mask)
+
+        # rescale the data 0 - 1
+        mask = 0.5 + mask * 0.5
+
+        # reorder dataset so 'horizontal_y' and 'horizontal_x' are the final dimensions
+        labels_orig = self.dimension_labels
+        labels = list(labels_orig)
+
+        labels.remove('horizontal_y')
+        labels.remove('horizontal_x')
+        labels.append('horizontal_y')
+        labels.append('horizontal_x')
+
+
+        if in_place == True:
+            self.reorder(labels)
+            numpy.multiply(self.array, mask, out=self.array)
+            self.reorder(labels_orig)
+
+        else:
+            image_data_out = self.copy()
+            image_data_out.reorder(labels)
+            numpy.multiply(image_data_out.array, mask, out=image_data_out.array)
+            image_data_out.reorder(labels_orig)
+
+            return image_data_out
+
+
+class VectorData(DataContainer):
+    '''DataContainer to contain 1D array'''
+
+    @property
+    def geometry(self):
+        return self._geometry
+
+    @geometry.setter
+    def geometry(self, val):
+        self._geometry = val
+
+    @property
+    def dimension_labels(self):
+        if hasattr(self,'geometry'):
+            return self.geometry.dimension_labels
+        else:
+            return self._dimension_labels
+
+    @dimension_labels.setter
+    def dimension_labels(self, val):
+        if hasattr(self,'geometry'):
+            self.geometry.dimension_labels = val
+
+        self._dimension_labels = val
+
+    def __init__(self, array=None, **kwargs):
+        self.geometry = kwargs.get('geometry', None)
+
+        dtype = kwargs.get('dtype', numpy.float32)
+
+        if self.geometry is None:
+            if array is None:
+                raise ValueError('Please specify either a geometry or an array')
+            else:
+                if len(array.shape) > 1:
+                    raise ValueError('Incompatible size: expected 1D got {}'.format(array.shape))
+                out = array
+                self.geometry = VectorGeometry(array.shape[0], **kwargs)
+                self.length = self.geometry.length
+        else:
+            self.length = self.geometry.length
+
+            if array is None:
+                out = numpy.zeros((self.length,), dtype=dtype)
+            else:
+                if self.length == array.shape[0]:
+                    out = array
+                else:
+                    raise ValueError('Incompatible size: expecting {} got {}'.format((self.length,), array.shape))
+        deep_copy = True
+        # need to pass the geometry, othewise None
+        super(VectorData, self).__init__(out, deep_copy, self.geometry.dimension_labels, geometry = self.geometry)
+
+
+class VectorGeometry(object):
+    '''Geometry describing VectorData to contain 1D array'''
+    RANDOM = 'random'
+    RANDOM_INT = 'random_int'
```

Comment:
These should have been replaced with labels.

Reply:
done (18e2609)